plugins/modules/gitlab_merge_request_rule.py

```diff
@@ -0,0 +1,565 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Based on code:
+# Copyright (c) 2023, Ondrej Zvara (ozvara1@gmail.com)
+# Copyright (c) 2021, Lennert Mertens (lennert@nubera.be)
+# Copyright (c) 2021, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2015, Werner Dijkerman (ikben@werner-dijkerman.nl)
+# Copyright (c) 2019, Guillaume Martinez (lunik@tiwabbit.fr)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+from __future__ import absolute_import, division, print_function
+
+__metaclass__ = type
+
+DOCUMENTATION = r"""
+module: gitlab_merge_request_rule
+short_description: Manage Gitlab Project merge request rules
+version_added: 3.4.0
+description:
+  - Managing Gitlab Project merge request rules (requires Enterprise/Ultimate Edition).
+author:
+  - mil1i (!UNKNOWN) [liebhaber_sgl@gwto.me]
+requirements:
+  - python-gitlab >= 2.6.0
+extends_documentation_fragment:
+  - community.general.auth_basic
+  - community.general.gitlab
+  - community.general.attributes
+
+attributes:
+  check_mode:
+    support: full
+  diff_mode:
+    support: none
+
+options:
+  state:
+    description:
+      - Create or delete protected branch.
+    default: present
+    type: str
+    choices: ["present", "absent"]
+  project_group:
+    description:
+      - The name/path of the group the project resides in.
+    required: true
+    type: str
+  project_name:
+    description:
+      - The name of the the project/repository.
+    required: true
+    type: str
+  rule_name:
+    description:
+      - The name of the custom merge request rule to manage.
+    required: true
+    type: str
+  approvals_required:
+    description:
+      - The number of approvals required before the rule passes to allow a merge.
+    required: true
+    type: int
+  usernames:
+    description:
+      - List of usernames of who can approve merge requests for this rule.
+    type: list
+    elements: str
+  user_ids:
+    description:
+      - List of user_ids of who can approve merge requests for this rule.
+    type: list
+    elements: int
+  groups:
+    description:
+      - List of group ids of members who can approve merge requests for this rule.
+    type: list
+    elements: int
+  protected_branch:
+    description:
+      - A protected branch name or branch id for this rule to apply to.
+      - Defaults to ALL branches if none specified.
+    type: str
+  applies_to_all_protected_branches:
+    description:
+      - Flag to enable applying this rule to ALL protected branches.
+      - Ignores "protected_branches" if this flag is set to True.
```

Comment:
```suggestion
      - Ignores O(protected_branches) if this flag is set to V(true).
```
Why does `protected_branches` have the special value `none` (unset) with the same meaning? I would remove that, and make `protected_branches` required unless this parameter is set to `true`. (And if this parameter is set to `true`, disallow specifying `protected_branches`.)